### PR TITLE
refactor(Data Export Controller): Use strategy pattern to break up class

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Changes
- Use [Strategy pattern](https://refactoring.guru/design-patterns/strategy) to refactor dataExportController
   - Create DataExportStrategy interface, DataExportContext class and AbstractDataExportStrategy class
   - Convert Discussion, Event, Notebook, Notification, OneWorkgroupPerRow, Raw, StudentAsset, StudentWork exports to strategies. They extend AbstractDataExportStrategy.

## Test
- All data exports work as before

### CODE REVIEW

File: src/assets/wise5/classroomMonitor/dataExport/strategies/OneWorkgroupPerRowDataExportStrategy.ts [hunk 1]
```diff
@@ -0,0 +1,625 @@
+import { DataExportContext } from '../DataExportContext';
+import DataExportController from '../dataExportController';
+import { AbstractDataExportStrategy } from './AbstractDataExportStrategy';
+import { DataExportStrategy } from './DataExportStrategy';
```
Comment (geoffreykwan) [inline src/assets/wise5/classroomMonitor/dataExport/strategies/OneWorkgroupPerRowDataExportStrategy.ts]: Remove unnecessary imports.

**Review by geoffreykwan — COMMENTED** [review]
All the exports seem to work properly.

It looks like for the component specific exports, only Discussion was updated to use the export strategy pattern. Do we want to update all the other component specific exports now too?

https://github.com/WISE-Community/WISE-Client/blob/f3d2c4f4ad9a8bceee1d7e82ee00adc8cbce8f91/src/assets/wise5/classroomMonitor/dataExport/dataExportController.ts#L985-L1024

File: src/assets/wise5/classroomMonitor/dataExport/strategies/OneWorkgroupPerRowDataExportStrategy.ts (hunk 1 above)
Comment (hirokiterashima) [reply src/assets/wise5/classroomMonitor/dataExport/strategies/OneWorkgroupPerRowDataExportStrategy.ts]: Done.

**Comment by hirokiterashima** [comment]
Let's merge this PR as is so that we can proceed with upgrading the grading tool to Angular. We can work on extracting the rest of the component exports in a separate task.

**Review by geoffreykwan — APPROVED** [review]
Looks good.

**Pull request merged**